// frontity-chakra-theme/src/components/fetchmenu/footermenu.js
import { Box, PseudoBox, Stack } from "@chakra-ui/core";
import { styled } from "frontity";
import React from "react";
import FrontityLink from "../link";

const Link = styled(FrontityLink)`
  position: relative;
  color: #fff;
  text-decoration: none;

  &:after {
    transition: bottom ease 0.25s, background-color ease 0.25s;
    content: "";
    width: 100%;
    height: 2px;
    position: absolute;
    bottom: 0;
    left: 0;
    background: transparent;
  }

  &:hover {
    &:after {
      bottom: -5px;
      background-color: ${p => p.theme.colors.accent[400]};
    }
  }
`;

export const SiteMenu = props => (
  <Stack
    as="ul"
    listStyleType="none"
    align="left"
    direction="column"
    color="white"
    {...props}
  />
);

const SiteMenuItem = ({ link, ...props }) => (
  <PseudoBox
    as="li"
    color="white"
    fontSize="0.75rem"
    fontWeight="medium"
    // fontFamily="Kelson"
    textTransform="uppercase"
    position="relative"
    cursor="pointer"
    {...props}
  >
    <Link link={link}>{props.children}</Link>
  </PseudoBox>
);

const FooterNav = ({ menu, ...props }) => (
  <Box as="nav" width="100%" display={{ base: "block", lg: "block" }} {...props}>
    <SiteMenu>
      {menu.map(([name, link]) => (
        <SiteMenuItem key={name} link={link}>
          {name}
        </SiteMenuItem>
      ))}
    </SiteMenu>
  </Box>
);

export default FooterNav;
