// frontity-chakra-theme/src/components/header/social-menu.js
import { Box, PseudoBox, VisuallyHidden, Button } from "@chakra-ui/core";
import React from "react";
import { SiteMenu } from "./navigation";
import {
  IoLogoTwitter,
  IoLogoPinterest,
  IoLogoFacebook,
  IoLogoInstagram
} from "react-icons/io";

import {FaWhatsapp, FaPhone} from "react-icons/fa";
import { faPhone } from '@fortawesome/free-solid-svg-icons';

import Link from "../link";
// import SiteMenuIcons from "./header-icons";

// warning for showSocialLinks and menu.length
export const SocialMenu = ({ menu, ...props }) => (
  <SiteMenu spacing="40px" ml="24px" position={{ sm: "relative" }} {...props}>
    {menu.map(([name, link]) => {
      const SocialIcon = icons[name];
      return (
        <SocialMenuItem key={name} label={name} link={link} icon={SocialIcon} />
      );
    })}
  </SiteMenu>
);

const SocialMenuItem = ({ icon, label, link, ...props }) => (
  <PseudoBox
    color="accent.400"
    transition="all 0.3s"
    _hover={{ color: "white" }}
    as="li"
    listStyleType="none"
    margin="auto"
    {...props}
    
  >
    <Box >
      <Box as={icon} size="24px" />
      {/* <Box>{link}</Box> */}
    </Box>
  </PseudoBox>
);

const icons = {
  twitter: IoLogoTwitter,
  pinterest: IoLogoPinterest,
  facebook: IoLogoFacebook,
  instagram: IoLogoInstagram,
  whatsapp: FaWhatsapp,
  phone: FaPhone
};



const SocialNav = ({ menu, ...props }) => (
  <Box ml="auto" display={{ base: "none", lg: "block" }} {...props}>
    <SocialMenu menu={menu} />
    {/* <SiteMenuIcons /> */}
  </Box>
);

export default SocialNav;
